src/com/facebook/buck/jvm/groovy/GroovyTestDescription.java

```diff
@@ -0,0 +1,193 @@
+import com.facebook.buck.jvm.java.TestType;
+import com.facebook.buck.model.BuildTarget;
+import com.facebook.buck.parser.NoSuchBuildTargetException;
+import com.facebook.buck.rules.BuildRule;
+import com.facebook.buck.rules.BuildRuleParams;
+import com.facebook.buck.rules.BuildRuleResolver;
+import com.facebook.buck.rules.BuildRuleType;
+import com.facebook.buck.rules.BuildRules;
+import com.facebook.buck.rules.BuildTargetSourcePath;
+import com.facebook.buck.rules.Description;
+import com.facebook.buck.rules.Hint;
+import com.facebook.buck.rules.Label;
+import com.facebook.buck.rules.SourcePath;
+import com.facebook.buck.rules.SourcePathResolver;
+import com.facebook.buck.rules.TargetGraph;
+import com.facebook.buck.util.HumanReadableException;
+import com.facebook.infer.annotation.SuppressFieldNotInitialized;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableSortedSet;
+import com.google.common.collect.Iterables;
+
+import java.nio.file.Path;
+import java.util.logging.Level;
+
+public class GroovyTestDescription implements Description<GroovyTestDescription.Arg> {
+
+  public static final BuildRuleType TYPE = BuildRuleType.of("groovy_test");
+
+  private final GroovyBuckConfig groovyBuckConfig;
+  private final JavacOptions defaultJavacOptions;
+  private final Optional<Long> defaultTestRuleTimeoutMs;
+  private final Optional<Path> testTempDirOverride;
+
+  public GroovyTestDescription(
+      GroovyBuckConfig groovyBuckConfig,
+      JavacOptions defaultOptions,
+      Optional<Long> defaultTestRuleTimeoutMs,
+      Optional<Path> testTempDirOverride) {
+    this.groovyBuckConfig = groovyBuckConfig;
+    this.defaultJavacOptions = defaultOptions;
+    this.defaultTestRuleTimeoutMs = defaultTestRuleTimeoutMs;
+    this.testTempDirOverride = testTempDirOverride;
+  }
+
+  @Override
+  public BuildRuleType getBuildRuleType() {
+    return TYPE;
+  }
+
+  @Override
+  public Arg createUnpopulatedConstructorArg() {
+    return new Arg();
+  }
+
+  @Override
+  public <A extends Arg> JavaTest createBuildRule(
+      TargetGraph targetGraph,
+      BuildRuleParams params,
+      BuildRuleResolver resolver,
+      A args) throws NoSuchBuildTargetException {
+    SourcePathResolver pathResolver = new SourcePathResolver(resolver);
+
+    BuildTarget abiJarTarget =
+        BuildTarget.builder(params.getBuildTarget())
+            .addFlavors(CalculateAbi.FLAVOR)
+            .build();
+
+    GroovycToJarStepFactory stepFactory = new GroovycToJarStepFactory(
+        groovyBuckConfig.getGroovyCompiler().get(),
+        args.extraGroovycArguments,
+        JavacOptionsFactory.create(
+            defaultJavacOptions,
+            params,
+            resolver,
+            pathResolver,
+            args
+        ));
+
+    JavaTest test =
+        resolver.addToIndex(
```

Comment:
I get 

`com.facebook.buck.util.HumanReadableException: Rule for target '//com/example/spock:passing' could not be resolved.`

The stack trace holds the answer:

<pre>
com.facebook.buck.util.HumanReadableException: Rule for target '//com/example/spock:passing' could not be resolved.

	at com.facebook.buck.rules.BuildRuleResolver.getRule(BuildRuleResolver.java:93)
	at com.facebook.buck.rules.SourcePathResolver$5.apply(SourcePathResolver.java:294)
	at com.facebook.buck.rules.SourcePathResolver$5.apply(SourcePathResolver.java:291)
	at com.google.common.collect.Iterators$8.transform(Iterators.java:799)
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:48)
	at com.google.common.collect.ImmutableList.copyOf(ImmutableList.java:271)
	at com.google.common.collect.ImmutableList.copyOf(ImmutableList.java:226)
	at com.google.common.collect.FluentIterable.toList(FluentIterable.java:373)
	at com.facebook.buck.rules.SourcePathResolver.filterBuildRuleInputs(SourcePathResolver.java:297)
	at com.facebook.buck.rules.SourcePathResolver.filterBuildRuleInputs(SourcePathResolver.java:301)
	at com.facebook.buck.jvm.java.CalculateAbi.of(CalculateAbi.java:65)
	at com.facebook.buck.jvm.groovy.GroovyTestDescription.createBuildRule(GroovyTestDescription.java:139)
	at com.facebook.buck.jvm.groovy.GroovyTestDescription.createBuildRule(GroovyTestDescription.java:49)
</pre>

So we need it to be in the resolver by the time we call `CalculateAbi.of`. I'll see if there's a clever way around this that could perhaps be applied to JavaTestDescription also, but I might save it for a separate commit.